### PR TITLE
Feature/dirbust

### PR DESCRIPTION
**Pull request by cobyge**
Added dirbust module.

### CODE REVIEW

File: IPScanner/Modules/dirbust.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
+import os
+import asyncio
+import itertools
+
+from IPScanner.Core import consts
+from IPScanner.Modules.base_module import BaseModule
+
+# Local Consts
+ERROR_CODES_TO_CATCH = [200, 401, 403, 405]
```
Comment (ofer1katz) [inline IPScanner/Modules/dirbust.py:9]: status codes

File: IPScanner/Modules/dirbust.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    def __init__(self, args):
+        """Init function.
+        Check for the specified dirlist.
+        -S --small for using the small dirlist contained in the project
+        -L --large for using the large dirlist contained in the project
+        -D --dirlist for using a custom dirlist file
+        -E --errors for using a custom error codes
+        """
+        super(DirBust, self).__init__(args)
+        self.url_to_scan = args.url_scheme + '://' + (args.domain or args.ip)
```
Comment (ofer1katz) [inline IPScanner/Modules/dirbust.py:40]: port?

File: IPScanner/Modules/dirbust.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        self.error_codes = [int(x) for x in args.errors.split(',')] if args.errors else ERROR_CODES_TO_CATCH
+        self.dirlist = args.large or args.small or args.dirlist
+        
+        if not os.path.exists(self.dirlist):
+            raise ValueError("Dirlist file does not exist!")
```
Comment (ofer1katz) [inline IPScanner/Modules/dirbust.py:45]: wordlist

File: IPScanner/Modules/dirbust.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+    async def send_request(self, group):
+        dirs = {}
+        
+        for directory in group:
+            search_dir = self.url_to_scan + '/{}'.format(directory)
+            try:
+                response = self.requests_wrapper.get(search_dir)
+            except Exception as e:
+                print (str(e))
```
Comment (ofer1katz) [inline IPScanner/Modules/dirbust.py:59]:
```suggestion
                print(str(e))
```

File: IPScanner/Modules/dirbust.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+            if response.status_code in self.error_codes:
+                dirs.update({search_dir:response.status_code})
```
Comment (ofer1katz) [inline IPScanner/Modules/dirbust.py:62]:
```suggestion
                dirs.update({search_dir: response.status_code})
```

File: IPScanner/Modules/dirbust.py [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+        
+    async def dirbust(self):
+        """This functions iterates on the dirlist file and tries to access it."""
+        dirs_busted = {}
+        async_tasks = []
+
+        with open(self.dirlist, 'r') as dirlist_file:
+            dir_list = dirlist_file.read().splitlines()
+
+        grouped_dirlist = list(self.dirlist_grouper(int(len(dir_list) / 10), dir_list))
```
Comment (ofer1katz) [inline IPScanner/Modules/dirbust.py:74]: how about the name "wordlist_groups"?

File: IPScanner/Modules/dirbust.py [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+        for group in grouped_dirlist:
+            task = asyncio.create_task(self.send_request(group))
+            async_tasks.append(task)
+
+        for task in async_tasks:
+            try:
+                dirs_busted.update(await task)
+            except Exception as e:
+                print (str(e))
```
Comment (ofer1katz) [inline IPScanner/Modules/dirbust.py:83]:
```suggestion
                print(str(e))
```

File: IPScanner/Modules/dirbust.py (hunk 1 above)
Comment (itay1996g) [reply IPScanner/Modules/dirbust.py:9]: Done

File: IPScanner/Modules/dirbust.py (hunk 6 above)
Comment (itay1996g) [reply IPScanner/Modules/dirbust.py:74]: Done

File: IPScanner/Modules/dirbust.py (hunk 3 above)
Comment (itay1996g) [reply IPScanner/Modules/dirbust.py:45]: Done

File: IPScanner/Modules/dirbust.py (hunk 2 above)
Comment (cobyge) [reply IPScanner/Modules/dirbust.py:40]: Added

File: IPScanner/Core/helper_functions.py [hunk 8]
```diff
@@ -99,6 +101,22 @@ def strip_url_protocol(args):
     return
 
 
+def strip_trailing_port(args):
+    if args.ip:
+        split_port = args.ip.split(":")
```
Comment (ofer1katz) [inline IPScanner/Core/helper_functions.py:106]: extract port with regex or at least comment that this function has to run after strip_url_protocol

File: IPScanner/Modules/openservices.py [hunk 9]
```diff
@@ -21,16 +29,16 @@ def __init__(self, args):
         super(OpenServicesScanner, self).__init__(args)
         self.open_services = []
         self.url_to_scan = args.url_scheme + '://' + args.domain
```
Comment (ofer1katz) [inline IPScanner/Modules/openservices.py:31]: port

File: IPScanner/Core/helper_functions.py (hunk 8 above)
Comment (cobyge) [reply IPScanner/Core/helper_functions.py:106]: Commented.  Regex isn't a real fix either.

File: IPScanner/Modules/openservices.py (hunk 9 above)
Comment (cobyge) [reply IPScanner/Modules/openservices.py:31]: Done

**Pull request merged**